### PR TITLE
Renesas mbedcrypto migration

### PR DESCRIPTION
**Pull request by michaelthomasj**
Notes:
## Description
Add support for Renesas vendor type keys. 

## Status
**READY**

## Requires Backporting
NO  

## Migrations
NO

## Additional comments
Any additional information that could be of interest

## Todos
- [x] Tests
- [x] Documentation
- [x] Changelog updated
- [x] Backported

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

File: include/psa/crypto_values.h [hunk 1]
```diff
@@ -381,6 +381,15 @@
  */
 #define PSA_KEY_TYPE_AES                            ((psa_key_type_t)0x2400)
 
+#if 0
+/** Whether a key type is AES. */
+#define PSA_KEY_TYPE_IS_AES(type) (((type) == PSA_KEY_TYPE_AES) != 0)
+#endif
+
+/** Whether a key type is AES; plaintext or wrapped. */
```
Comment (michaelthomasj) [inline include/psa/crypto_values.h:384]: Change to logical OR

File: include/psa/crypto_values.h [hunk 2]
```diff
@@ -415,9 +424,23 @@
...
+#endif
+
+/** Whether a key type is an RSA key pair; standard or vendor. */
+#define PSA_KEY_TYPE_IS_RSA_KEY_PAIR(type)  							\
+	((type == PSA_KEY_TYPE_RSA_KEY_PAIR) || \
+	 (type == (PSA_KEY_TYPE_RSA_KEY_PAIR | PSA_KEY_TYPE_VENDOR_FLAG)))
+
+
+/** Whether a key type is an RSA key (pair or public-only) standard or vendor. */
+#define PSA_KEY_TYPE_IS_RSA(type)                                       \
```
Comment (michaelthomasj) [inline include/psa/crypto_values.h:430]: Change to logical OR

File: include/psa/crypto_values.h [hunk 3]
```diff
@@ -1577,7 +1621,15 @@
  * by other lifetime values as implementation-specific extensions.
  * See ::psa_key_lifetime_t for more information.
  */
+
 #define PSA_KEY_LIFETIME_PERSISTENT             ((psa_key_lifetime_t)0x00000001)
+#define PSA_KEY_LIFETIME_PERSISTENT_FLAG        ((psa_key_lifetime_t)PSA_KEY_LIFETIME_PERSISTENT)
+//#define PSA_KEY_LIFETIME_IS_VOLATILE(lifetime)  (((lifetime) & PSA_KEY_LIFETIME_PERSISTENT_FLAG) == 0)
```
Comment (michaelthomasj) [inline include/psa/crypto_values.h]: remove commented out code

File: include/psa/crypto_values.h [hunk 4]
```diff
@@ -1577,7 +1621,15 @@
  * by other lifetime values as implementation-specific extensions.
  * See ::psa_key_lifetime_t for more information.
  */
+
 #define PSA_KEY_LIFETIME_PERSISTENT             ((psa_key_lifetime_t)0x00000001)
+#define PSA_KEY_LIFETIME_PERSISTENT_FLAG        ((psa_key_lifetime_t)PSA_KEY_LIFETIME_PERSISTENT)
```
Comment (michaelthomasj) [inline include/psa/crypto_values.h]: remove this unused macro

File: library/psa_crypto.c [hunk 5]
```diff
@@ -783,8 +792,16 @@ static psa_status_t psa_load_ecp_representation( psa_key_type_t type,
     mbedtls_ecp_keypair_init( ecp );
 
     /* Load the group. */
-    grp_id = mbedtls_ecc_group_of_psa( PSA_KEY_TYPE_ECC_GET_FAMILY( type ),
-                                       curve_size );
+#if defined (MBEDTLS_PSA_CRYPTO_ACCEL_DRV_C) && defined(MBEDTLS_ECP_ALT)
+    if (PSA_KEY_TYPE_IS_VENDOR_DEFINED(type))
+    {
+    	grp_id = mbedtls_ecc_group_of_psa( PSA_KEY_TYPE_ECC_GET_FAMILY( type ), PSA_ECC_BYTES_VENDOR_RAW(curve_size) );
```
Comment (michaelthomasj) [inline library/psa_crypto.c:798]: Consider renaming _RAW

File: include/psa/crypto_values.h (hunk 4 above)
Comment (michaelthomasj) [reply include/psa/crypto_values.h]: Done

File: include/psa/crypto_values.h (hunk 3 above)
Comment (michaelthomasj) [reply include/psa/crypto_values.h]: Done

File: include/psa/crypto_values.h (hunk 1 above)
Comment (michaelthomasj) [reply include/psa/crypto_values.h:384]: Done

File: include/psa/crypto_values.h (hunk 2 above)
Comment (michaelthomasj) [reply include/psa/crypto_values.h:430]: Done

File: library/psa_crypto.c [hunk 6]
```diff
@@ -2106,7 +2146,7 @@ static psa_status_t psa_finish_key_creation(
     (void) driver;
 
 #if defined(MBEDTLS_PSA_CRYPTO_STORAGE_C)
-    if( ! PSA_KEY_LIFETIME_IS_VOLATILE( slot->attr.lifetime ) )
+    if( PSA_KEY_LIFETIME_IS_PERSISTENT( slot->attr.lifetime ) )
```
Comment (michaelthomasj) [inline library/psa_crypto.c]: This can possible be reverted. Check.

File: library/psa_crypto.c [hunk 7]
```diff
@@ -4901,8 +4964,28 @@ static psa_status_t psa_aead_setup( aead_operation_t *operation,
...
+        }
+
+        temp_keytype = (psa_key_type_t)(operation->slot->attr.type & ~PSA_KEY_TYPE_VENDOR_FLAG);
+    }
+    else
+#endif /* MBEDTLS_PSA_CRYPTO_ACCEL_DRV_C */
+     {
+        temp_keytype = (psa_key_type_t)(operation->slot->attr.type);
+    }
     operation->cipher_info =
```
Comment (michaelthomasj) [inline library/psa_crypto.c:4987]: Move this to inside the function

Comment (michaelthomasj) [reply library/psa_crypto.c:4987]: This causes more issues and changes. So not doing it.

File: library/psa_crypto.c (hunk 5 above)
Comment (michaelthomasj) [reply library/psa_crypto.c:798]: Not doing.

Comment (aniruddhabajaj-renesas) [reply library/psa_crypto.c:798]: discussed, not needed to be updated for this ticket.

**Pull request merged**